src/main/java/com/owncloud/android/lib/resources/files/NcSearchMethod.java

```diff
@@ -287,17 +287,7 @@ private Document createQuery(String searchQuery) {
         } else {
             if (searchType == SearchRemoteOperation.SearchType.GALLERY_SEARCH) {
                 Element and = query.createElementNS(DAV_NAMESPACE, "d:and");
-                Element lessThan = query.createElementNS(DAV_NAMESPACE, "d:eq");
-                Element lastModified = query.createElementNS(NAMESPACE_OC, "oc:owner-id");
-                Element literal = query.createElementNS(DAV_NAMESPACE, "d:literal");
-                Element prop = query.createElementNS(DAV_NAMESPACE, "d:prop");
-                prop.appendChild(lastModified);
-                literal.setTextContent(String.valueOf(userId));
 
-                lessThan.appendChild(prop);
-                lessThan.appendChild(literal);
-
-                and.appendChild(lessThan);
```

Comment:
We should only delete those lines starting from NC22 and above. Is there a way to do this? (E.g. get the server version at this place and add a condition...

Reply:
Not directly inside `NcSearchMethod` as far as I know. However `NcSearchMethod` is called only in [`SearchOperation`](http://github.com/nextcloud/android-library/blob/ff59951f63124d2ed7777d312f9b04558f1eb991/src/main/java/com/owncloud/android/lib/resources/files/SearchRemoteOperation.java#L98-L98), so theoretically you could call `GetCapabilitiesRemoteOperation`inside that, and pass the resulting version to `NcSearchMethod`.

Example of calling `GetCapabilitiesRemoteOperation` inside another operation: http://github.com/nextcloud/android-library/blob/ff59951f63124d2ed7777d312f9b04558f1eb991/src/main/java/com/owncloud/android/lib/common/accounts/ExternalLinksOperation.java#L79-L79. You should then be able to use `capability.getVersion().isNewerOrEqual(NextcloudVersion.nextcloud_22)` to check the server version.

Reply:
By the way, I think you should also probably skip the creation of `and`, and just append `equalsElement` to `whereElement`, like in the `else` block.

Reply:
Thanks for the review!
I hope I fixed everything in https://github.com/nextcloud/android-library/pull/782/commits/c0d96261d18b23e78ec120729b18f2e365f84274
But I must admit that I still understand the syntax only partially.

Reply:
seems like the compiling has failed. What did I do wrong?

Reply:
Calling get remote capability all the time is a bit overhead, I think.
Just add the capability to SearchRemoteOperation, so that it is required.